probe-rs/src/probe/blackmagic/arm.rs

```diff
@@ -324,6 +329,28 @@ impl ArmProbeInterface for BlackMagicProbeArmDebug {
     }
 
     fn reinitialize(&mut self) -> Result<(), ArmError> {
+        let sequence = self.sequence.clone();
+        let dp = self.current_debug_port();
+
+        // Switch to the correct mode
+        sequence.debug_port_setup(&mut *self.probe, dp)?;
+
+        if let Err(e) = sequence.debug_port_connect(&mut *self.probe, dp) {
+            tracing::warn!("failed to switch to DP {:x?}: {}", dp, e);
+
+            // Try the more involved debug_port_setup sequence, which also handles dormant mode.
+            sequence.debug_port_setup(&mut *self.probe, dp)?;
+        }
+
+        self.debug_port_start(dp).unwrap();
+
+        self.access_ports = valid_access_ports(self, DpAddress::Default)
+            .into_iter()
+            .collect();
+        self.access_ports.iter().for_each(|addr| {
+            tracing::debug!("AP {:#x?}", addr);
+        });
```

Comment:
You can probably `inspect` the iterator while you're collecting the ports.

I'd probably also like to see that `unwrap` handled, or explained.

Reply:
I got rid of the `.unwrap()` in both places.

Additionally, I converted it to use `inspect()`, and converted the originating stlink code to also use `inspect()`.